File: Go/Go_1000-/1422.go
// Go
// leetcode 1422
// https://leetcode.cn/problems/maximum-score-after-splitting-a-string/

func maxScore(s string) (res int) {
	n := len(s)
	cur := 0
	for i := 0; i < n - 1; i++ {
		if s[i] == '0' {
			cur++
		}
		res = max(res, 2 * cur + n - i - 1)
	}
	return res - cur - int('1' - s[n - 1])
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: cpp/cpp_1000-/1422.cc
// C++
// leetcode 1422
// https://leetcode.cn/problems/maximum-score-after-splitting-a-string/

class Solution {
public:
    int maxScore(string s) {
        int n = s.size(), cnt = 0, res = - n;
        for(int i = 0; i < n - 1; i++) {
            cnt += (s[i] == '0' ? 1 : 0);
            res = max(res, 2 * cnt + n - i - 1);
        }
        res -= cnt + (s[n-1] == '0' ? 1 : 0);
        return res;
    }
};